awx/ui_next/src/screens/Host/Host.jsx

```diff
@@ -117,62 +196,99 @@ class Host extends Component {
         </PageSection>
       );
     }
-
     return (
-      <PageSection>
-        <Card className="awx-c-card">
-          {cardHeader}
-          <Switch>
-            <Redirect from="/hosts/:id" to="/hosts/:id/details" exact />
-            {host && (
-              <Route
-                path="/hosts/:id/edit"
-                render={() => <HostEdit match={match} host={host} />}
-              />
-            )}
-            {host && (
-              <Route
-                path="/hosts/:id/details"
-                render={() => <HostDetail match={match} host={host} />}
-              />
-            )}
-            {host && (
-              <Route
-                path="/hosts/:id/facts"
-                render={() => <HostFacts host={host} />}
-              />
-            )}
-            {host && (
-              <Route
-                path="/hosts/:id/groups"
-                render={() => <HostGroups host={host} />}
-              />
-            )}
-            {host && (
+      <>
+        <PageSection
+          css={`
+            ${location.pathname.startsWith('/inventories')
+              ? 'padding: 0'
+              : 'null'}
+          `}
+        >
+          <Card className="awx-c-card">
+            {cardHeader}
+            <Switch>
+              <Redirect from="/hosts/:id" to="/hosts/:id/details" exact />
+              {host && (
+                <Route
+                  path={[
+                    '/hosts/:id/edit',
+                    '/inventories/inventory/:id/hosts/:hostId/edit',
+                  ]}
+                  render={() => <HostEdit match={match} host={host} />}
+                />
+              )}
+              {host && (
+                <Route
+                  path={[
+                    '/hosts/:id/details',
+                    '/inventories/inventory/:id/hosts/:hostId/details',
+                  ]}
+                  render={() => (
+                    <HostDetail
+                      match={match}
```

Comment:
do we pass match in other places?  I feel like the better pattern would be to wrap the HostDetail.jsx component in the react-router provider, but I'm not sure if this is what we do everywhere, and that's why this is done this way.

This does feel like a lot of props...that suggestion of moving that stuff to the HostDetail function would make this simpler.

Reply:
no, we don't pass match in other places I have already removed that, and have been able to reduce the number of props going to Host Details to 2(host, and onUpdateHost, which is a function that HostDetails uses to update host.)